internal/provider/client.go

```diff
@@ -68,14 +74,46 @@ func readApiEndpoint(client ProviderHTTPClient) (string, diag.Diagnostics) {
 				fmt.Sprintf("Unable to parse AAP API endpoint response: %s", string(body)),
 				fmt.Sprintf("Unexpected error: %s", err.Error()),
 			)
-			return "", diags
+			return "", "", diags
+		}
+		controllerEndpoint = response.CurrentVersion
+	} else {
+		if len(response.CurrentVersion) == 0 {
+			diags.AddError("Unable to determine API Endpoint", "The controller endpoint is missing from response")
+			return "", "", diags
 		}
+		controllerEndpoint = response.CurrentVersion
 	}
-	if len(response.CurrentVersion) == 0 {
-		diags.AddError("Unable to determine API Endpoint", "The controller endpoint is missing from response")
-		return "", diags
+
+	if len(response.Apis.EDA) > 0 {
+		body, diags = client.Get(response.Apis.EDA)
+		if diags.HasError() {
+			return "", "", diags
+		}
+		err = json.Unmarshal(body, &response)
+		if err != nil {
+			diags.AddError(
+				fmt.Sprintf("Unable to parse EDA API endpoint response: %s", string(body)),
+				fmt.Sprintf("Unexpected error: %s", err.Error()),
+			)
+			return "", "", diags
+		}
+
+		// URL Parse because the EDA endpoint's current_version gives a full URL instead of a URL path
```

Comment:
Interesting. So the format of `current_version` from controller is not consistent with EDA? I notice we don't need to `url.Parse()` that response from controller

Reply:
Yes I was surprised to learn they have different behavior.

Here is what the controller endpoint response looks like:
```json
{
  "description": "AWX REST API",
  "current_version": "/api/controller/v2/",
  "available_versions": {
    "v2": "/api/controller/v2/"
  },
  "custom_logo": "",
  "custom_login_info": "",
  "login_redirect_override": ""
}
```

Here is what the EDA endpoint response looks like:
```json
{
  "current_version": "http://localhost:44925/api/eda/v1/",
  "available_versions": {
    "v1": "http://localhost:44925/api/eda/v1/"
  }
}
```